apps/openmw/mwmechanics/spellcasting.cpp

```diff
@@ -32,41 +32,23 @@ namespace MWMechanics
 {
     ESM::Skill::SkillEnum spellSchoolToSkill(int school)
     {
-        static std::map<int, ESM::Skill::SkillEnum> schoolSkillMap; // maps spell school to skill id
-        if (schoolSkillMap.empty())
+        static const std::map<int, ESM::Skill::SkillEnum> schoolSkillMap
         {
-            schoolSkillMap[0] = ESM::Skill::Alteration;
-            schoolSkillMap[1] = ESM::Skill::Conjuration;
-            schoolSkillMap[3] = ESM::Skill::Illusion;
-            schoolSkillMap[2] = ESM::Skill::Destruction;
-            schoolSkillMap[4] = ESM::Skill::Mysticism;
-            schoolSkillMap[5] = ESM::Skill::Restoration;
-        }
-
-        assert(schoolSkillMap.find(school) != schoolSkillMap.end());
-        return schoolSkillMap[school];
-    }
-
-    float calcEffectCost(const ESM::ENAMstruct& effect)
-    {
-        const ESM::MagicEffect* magicEffect = MWBase::Environment::get().getWorld()->getStore().get<ESM::MagicEffect>().find(effect.mEffectID);
-        return calcEffectCost(effect, magicEffect);
+            {0, ESM::Skill::Alteration}, {1, ESM::Skill::Conjuration}, {2, ESM::Skill::Destruction},
+            {3, ESM::Skill::Illusion}, {4, ESM::Skill::Mysticism}, {5, ESM::Skill::Restoration}
+        };
+        return schoolSkillMap.at(school);
```

Comment:
`school` is expected to have a value from 0 to 5. Looks like an array or vector would do the same.

Reply:
Ok, updated.